### ISSUE TITLE
[BUG] Terraform - Failed to get existing workspaces: containers.Client#ListBlobs

### ISSUE THREAD

**Issue by code-bucket-user** [microsoft/NubesGen#474]
**Describe the bug**
During the execution of gitops action, it is failure on **steps - name:** Apply Terraform configuration.
Terraform backend configured properly as per the standard. **steps -uses:** microsoft/nubesgen-actions/gitops-apply-terraform@v0.13.0. 
But it is throwing bellow error 

`**Error:** Failed to get existing workspaces: containers.Client#ListBlobs: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ContainerNotFound" Message="The specified container does not exist.\nRequestId:*****`    

Where as existing **backend configuration** running perfectly during _**manual execution**_ using `terraform init / terraform init -reconfigure`

**To Reproduce**
1. Create an Storage account.  
2. Create a Blob Storage Container.
3. Configure backend "azurerm" { 
    resource_group_name  = "rg-sample-dev-us"
    storage_account_name = "st001sampleus"
    container_name            = "sample-container-us-1"
    key                                = "terraform.tfstate"
    access_key                    = "********"
}

4. Run the gitops action. 

**Expected behavior**
Initializing the backend...

Successfully configured the backend "azurerm"! Terraform will automatically
use this backend unless the backend configuration changes.
Initializing modules...
                                                                                                                
Initializing provider plugins...
- Finding aztfmod/azurecaf versions matching "1.2.26"...
- Finding hashicorp/azurerm versions matching "~> 3.0"...
- Installing aztfmod/azurecaf v1.2.26...
- Installed aztfmod/azurecaf v1.2.26 (self-signed, key ID 3D457F4D69B172D6)
- Installing hashicorp/azurerm v3.78.0...
- Installed hashicorp/azurerm v3.78.0 (signed by HashiCorp)

**Screenshots**
Please find the attached screenshot
![Error_1](https://github.com/microsoft/NubesGen/assets/71180966/955f8c7c-128b-4cde-a507-66d5e13da251)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - Browser [e.g. chrome]

**Comment by jdubois** [comment]
What you describe is what everybody does with Terraform, then we have a specific configuration that I'm not sure you're aware of: we add a VNet to secure the access to the Blob storage.
As the Terraform state stores a lot of sensitive data (like the database password), we believe it's better not to have it available on the Internet directly, so we add this extra layer of security.
As you don't write about it in your issue, it seems to me you configured this directly. Can you check if you have a VNet securing your blob storage?

**Comment by code-bucket-user** [comment]
Yes VNet is configured. I follow the every steps those are mentioned in [docs.nubesgen.com](https://docs.nubesgen.com/what-is-nubesgen/overview/).

Not only V-Net along with subnet, network rule, other principal for permission, relevant role such as contributor, owner e.t.c configured properly. 

What I observed in gitops **workflow log** during `run export ARM_CLIENT_SECRET=$(echo $AZURE_CREDENTIALS | jq -r .clientSecret)` do` terraform init` taking wrong `resource_group_name` & `container_name` as `-backend-config`.  Although `terraform backend` is properly configured. As `resource_group_name` is part of the` inputs` so I able to override it within workflow yml, but unable to do the same for `container_name`. As becuase it is not taking any inputs. So only way to resolve it for working propose create a blob container name `tfstate`.  

If anything else required to configure, please let me know.
